### ISSUE TITLE
Attempt to fetch from allocatable variable INDEX_VERTICALAEROSOLSCONC when it is not allocated

### ISSUE THREAD

**Issue by amametjanov** [E3SM-Project/E3SM#4018]
On latest `maint-1.0` branch, `./cime/scripts/create_test SMS_D_Ld1.ne30_oECv3_ICG.A_WCYCL1850S_CMIP6.chrysalis_intel.allactive-v1cmip6` fails with
```
[164] forrtl: severe (408): fort: (8): Attempt to fetch from allocatable variable INDEX_VERTICALAEROSOLSCONC when it is not allocated
[164]
[164] Image              PC                Routine            Line        Source
[164] libpnetcdf.so.4.0  0000155548015892  for_emit_diagnost     Unknown  Unknown
[164] e3sm.exe           0000000006D4007F  seaice_column_mp_        8459  mpas_seaice_column.f90
[164] e3sm.exe           0000000006D36984  seaice_column_mp_        7824  mpas_seaice_column.f90
[164] e3sm.exe           0000000006BE89F3  seaice_column_mp_         228  mpas_seaice_column.f90
[164] e3sm.exe           0000000006DDD00A  seaice_initialize         102  mpas_seaice_initialize.f90
[164] e3sm.exe           0000000006B11B2E  seaice_core_mp_mp         255  mpas_seaice_core.f90
[164] e3sm.exe           0000000006B108EB  seaice_core_mp_se         145  mpas_seaice_core.f90
[164] e3sm.exe           00000000043041BB  ice_comp_mct_mp_i         522  ice_comp_mct.f90
[164] e3sm.exe           00000000004639E9  component_mod_mp_         269  component_mod.F90
[164] e3sm.exe           0000000000426BCF  cime_comp_mod_mp_        1198  cime_comp_mod.F90
[164] e3sm.exe           000000000045971F  MAIN__                     92  cime_driver.F90
[164] e3sm.exe           0000000000414FA2  Unknown               Unknown  Unknown
[164] libc-2.28.so       0000155544DFD6A3  __libc_start_main     Unknown  Unknown
[164] e3sm.exe           0000000000414EAE  Unknown               Unknown  Unknown
```

That array is allocated when `config_use_column_biogeochemistry=.true.`, but
```
$ grep -ir "config_use_column_biogeochemistry" mpassi_in 
 config_use_column_biogeochemistry = .false.
```

Relevant `CPP`-ed lines:
```
  224        ! initialize the column package tracer object
  225        call init_column_tracer_object(domain, ciceTracerObject)
  226 
  227        ! initialize the column package parameters
  228        call init_column_package_parameters(domain, ciceTracerObject)
  229 
  230     endif
  231 
  232   end subroutine seaice_init_column_physics_package_parameters
```
```
 7823     ! set the tracers indices
 7824     call init_column_package_tracer_indices(tracerObject)
 7825 
 7826     ! set the column parameters
 7827     call init_column_package_configs(domain)
 7828 
 7829   end subroutine init_column_package_parameters
```
```
 8459     call colpkg_init_tracer_indices(&
 8460          tracerObject % index_surfaceTemperature, &
 8461          tracerObject % index_iceEnthalpy, &
 8462          tracerObject % index_snowEnthalpy, &
 8463          tracerObject % index_iceSalinity, &
 8464          tracerObject % index_brineFraction, &
 8465          tracerObject % index_iceAge, &
 8466          tracerObject % index_firstYearIceArea, &
 8467          tracerObject % index_levelIceArea, &
 8468          tracerObject % index_levelIceVolume, &
 8469          tracerObject % index_pondArea, &
 8470          tracerObject % index_pondDepth, &
 8471          tracerObject % index_pondLidThickness, &
 8472          tracerObject % index_aerosols, &
 8473          tracerObject % index_verticalAerosolsConc, &
```
```
 4610     ! biogeochemistry
 4611     if (config_use_column_biogeochemistry) then
 4612 
 4613        allocate(tracerObject % index_LayerIndexToDataArray(nZBGCTracers))
 4614        allocate(tracerObject % index_LayerIndexToBioIndex(nZBGCTracers))
 4615 
 4616        ! set all indices for biogeochemistry including parent, ancestor and ancestor mask
 4617        call init_column_tracer_object_for_biogeochemistry(domain, tracerObject)
 4618 
 4619     endif
 4620 
 4621   end subroutine init_column_tracer_object
```

**Comment by njeffery** [comment]
@amametjanov : these errors have been corrected in the current version of mpas

**Comment by amametjanov** [comment]
@mark-petersen Can you please update `maint-1.0` branch and point at the correct hash? `mpas-source` is at hash [a66baa2](https://github.com/MPAS-Dev/MPAS-Model/tree/a66baa2b4ec4c784ecb073ac377b00e94f283f88) (for `e3sm/maint-1.0`) right now.

**Comment by ndkeen** [comment]
Does it make sense to fix this error without changing much else? Or is that possible? Ideally, want to stay BFB for the production runs.

**Comment by njeffery** [comment]
I believe this error appeared about a year ago with a compiler upgrade.  The fix is fairly straightforward and should be BFB.  If you decide to go this route, I'm aware of several other variables that will also need allocating.  And perhaps some other fixes...

**Comment by rljacob** [comment]
There should be a branch in MPAS corresponding to the maint-1.0 branch here and the fix should be done on that branch.

**Comment by wagmanbe** [comment]
Has a fix been found?  Or, @rljacob @amametjanov have you successfully pointed to another branch of MPAS? I have the same error in a A_WCYCL20TRS_CMIP6 compset on the maint-1.0 branch on chrysalis. The error only occurs after I use a modified aerosol emissions file, but perhaps the same bug is to blame. The mpas_source is still at hash a66baa2b.

**Comment by rljacob** [comment]
@jonbob can you add the fix for this issue to the maint-1.0 version of MPAS?

**Comment by jonbob** [comment]
@rljacob - let me see what I can do

**Comment by jonbob** [comment]
Fixed by [PR #4352](https://github.com/E3SM-Project/E3SM/pull/4352)
